### ISSUE TITLE
Feature Request: Option to hide default macOS app label 

### ISSUE THREAD

**Issue by langjacob518** [ejbills/DockDoor#140]
I think it would be a great addition to the app to be able to get rid of the default text label so that it doesn't obstruct the window preview (the text "Notion" in it's own bubble in the image below"). Tbh I think this would be great as the default setting for DockDoor since DockDoor can already list the app name in the top left of the window if desired. Love where the app is going! Thanks for your work. 

<img width="1091" alt="Screenshot 2024-07-10 at 4 59 47 PM" src="https://github.com/ejbills/DockDoor/assets/167139396/95558fab-252e-4b16-ad63-5cb44c5bf60b">

**Comment by ShlomoCode** [comment]
I wonder if this is even possible without changes in `Dock.app` (which would require the removal of macOS System Integrity Protection)
Do you know of any app out there that can do this?

**Comment by UnknownCrafts** [comment]
> I wonder if this is even possible without changes in `Dock.app` (which would require the removal of macOS System Integrity Protection) Do you know of any app out there that can do this?

I was researching about this yesterday and I did find a script that someone made, and I am pretty sure that it is possible to do without disabling SIP. Here is the link to the said script: https://github.com/vicneamt/dock_title (Original version), My Version: https://github.com/UnknownCrafts/dock_title (I am working on ironing out some bugs from the OG version). Basically, it edits the "~/Library/Preferences/com.apple.dock.plist" to hide the app titles. One issue I am facing right now is that if an app that is not a persistent dock app and the recent apps functionality is disabled, is running then for whatever reason the dock.plist file wouldn't have any mention of that app meaning we can't strip its title away. Also I think that DockDoor would have to consistently monitor the dock for any app changes so that it can keep on stripping app titles, at least that's what my understanding is so far.

**Comment by UnknownCrafts** [comment]
Ah, I actually didn't know about that. Maybe DockDoor could handle that if this feature is enabled? maybe by first hiding or unhiding the dock (I am not sure if that's enough to generate the plist)

**Comment by ejbills** [comment]
I don't think it can be done automatically. The only way we can modify the dock programmatically is via that plist file. Which came first, the chicken or the egg? We have that situation lol, we will need to instruct the user to do it manually. 

Like "to generate the plist file, close DockDoor, open your system settings, change the dock's magnification setting a few times and restart the app, it should work."

**Comment by ejbills** [comment]
Oh good point, tbh I'm not familiar with Apple Script.

**Comment by UnknownCrafts** [comment]
https://forums.developer.apple.com/forums/thread/98830 (Pray to Apple overlords that this isn't deprecated), seems like its also kinnda painless to integrate Applescript in swift apps. I have some experience with creating stuff with AppleScript, mainly thanks to creating shortcuts in Monarch.

**Comment by langjacob518** [comment]
I'm pretty sure DockView (paid app) has this ability. I know really nothing about coding, but I have a hunch that DockView "hijacks" the default macOS label to make their window previews. I'm basing this solely on how their window preview has the same style on bottom as the macOS default label, so I could be totally off-base here. But using DockView, I didn't have to disable SIP or do anything crazy. It just shipped like this. Here's a picture: 
<img width="882" alt="Screenshot 2024-07-11 at 10 44 57 AM" src="https://github.com/ejbills/DockDoor/assets/167139396/558c6e6d-74eb-4e2f-afe9-7c0a14a9871b">

And here's the link to the app (free trial if you want to test it out and see what it looks like): https://noteifyapp.com/dockview/

**Comment by ShlomoCode** [comment]
https://github.com/kcrawford/dockutil

**Comment by jacman23** [comment]
What other applications do is place the window in front of the name, so the dock name remains hidden.

**Comment by ejbills** [comment]
From ErickWei

Go to /Users/Your Name/Library/Preferences/
Open com.apple.dock.plist (It should open in Property List Editor)
In the Editor disclose Root >> persistent-apps >> 0 >> Tile Data
set file-label to empty

This hides the app title label but also causes DockDoor to not render its window.

**Comment by DarioLopes** [comment]
May I to the original request : hide app label only if there is a preview (just so we don't lose them on closed apps)

**Comment by ejbills** [comment]
Future versions will have a default dock padding of -25, along with the window preview open delay set to 0.1 if you have dock hide enabled to allow the dock frame to update. 

In my experience, this -25 window buffer covers the app label just enough for me to feel comfortable to raise the window level so that it covers the app label. It would be helpful if users could test with these values and let me know if if works on your configuration.

**Comment by ejbills** [comment]
With macOS 26 being a lot more transparent, simply covering the app label won't suffice.

**Comment by ejbills** [comment]
With macOS 26 being a lot more transparent, simply covering the app label won't suffice.
